// reactlet-html-json-edit/dist/html_json_edit.js
var React = React || require('react/addons');

module.exports = React.createClass({displayName: "exports",
    name: 'html-jsonedit',
    mixins: [getCommonMixin],
    
    // attribute definitions
    getAttributes: function() {
        var attributes = [
            { name:'name', type:'string', required:true, defaultValue:'', note:'field name' },
            { name:'dispatcher', type:'object', required:false, defaultValue:null, note:'flux dispatcher' },
            { name:'boxClass', type:'string', required:false, defaultValue:'', note:'container CSS class' },
            { name:'inputClass', type:'string', required:false, defaultValue:'form-control', note:'input element CSS class' },
            { name:'type', type:'string', required:false, defaultValue:'json', note:'input type' },
            { name:'label', type:'string', required:false, defaultValue:'', note:'input label' },
            { name:'value', type:'string', required:false, defaultValue:'', note:'input type' }
        ];
        return attributes;
    },
    
    getValue: function() {
        return this.state.value;
    },
    
    onChange: function(event) {
        var newValue = event.target.value;
        this.setState({ value:newValue });
        this.fire('change', [newValue]);
    },
    
    render: function() {
        var jsonValue = this.state.value;
        if (typeof this.state.value === 'object') {
            jsonValue = JSON.stringify(this.state.value);
        }
        // set content display
        var content = 
            React.createElement("div", {className:  "form-group html-json-edit-content-container" }, 
                React.createElement("label", null,  this.state.label, " (json)"), 
                React.createElement("textarea", {
                    name:  this.state.name, 
                    className:  this.state.inputClass, 
                    value:  jsonValue, 
                    onChange:  this.onChange}
                )
            );
        return (
            React.createElement("div", {className:  this.state.containerClassNames.join(' ') }, 
                 content 
            )
        );
    }
});
