// reactlet-html-web-edit/dist/html_web_edit.js
var React = React || require('react/addons');

module.exports = React.createClass({displayName: "exports",
    name: 'html-webedit',
    mixins: [getCommonMixin],
    
    // attribute definitions
    getAttributes: function() {
        var attributes = [
            { name:'name', type:'string', required:true, defaultValue:'', note:'field name' },
            { name:'dispatcher', type:'object', required:false, defaultValue:null, note:'flux dispatcher' },
            { name:'boxClass', type:'string', required:false, defaultValue:'', note:'container CSS class' },
            { name:'inputClass', type:'string', required:false, defaultValue:'form-control', note:'input element CSS class' },
            { name:'type', type:'string', required:false, defaultValue:'json', note:'input type' },
            { name:'label', type:'string', required:false, defaultValue:'', note:'input label' },
            { name:'value', type:'string', required:false, defaultValue:'', note:'input type' }
        ];
        return attributes;
    },
    
    getValue: function() {
        return this.state.value;
    },
    
    onChange: function(event) {
        var newValue = event.target.value;
        this.setState({ value:newValue });
        this.fire('change', [newValue]);
    },
    
    componentDidMount: function() {
        
        console.log('>>> componentDidMount - ready for summernote');
        
        // set summernote for targeted dom elements
        
        /*
        $('#summernote').summernote({
            height: 300,                 // set editor height
            minHeight: null,             // set minimum height of editor
            maxHeight: null,             // set maximum height of editor
            focus: false,                 // set focus to editable area after initializing summernote
            onImageUpload: function(files, editor, welEditable) {
                console.log('image upload:', files, editor, welEditable);
                sendFile(files[0], editor, welEditable);
            }
        });
        */
    },
    
    render: function() {
        var displayValue = this.state.value;
        // set content display
        var content = 
            React.createElement("div", {className:  "form-group html-web-edit-content-container" }, 
                React.createElement("label", null,  this.state.label, " (html)"), 
                React.createElement("textarea", {
                    name:  this.state.name, 
                    className:  this.state.inputClass, 
                    value:  displayValue, 
                    onChange:  this.onChange}
                )
            );
        return (
            React.createElement("div", {className:  this.state.containerClassNames.join(' ') }, 
                 content 
            )
        );
    }
});
