// reactlet-html-text-edit/dist/html_text_edit.js
var React = React || require('react/addons');

// HtmlTextEdit component
var HtmlTextEdit = React.createClass({displayName: "HtmlTextEdit",
    name: 'html-textedit',
    mixins: [getCommonMixin],
    
    // attribute definitions
    getAttributes: function() {
        var attributes = [
            { name:'name', type:'string', required:true, defaultValue:'', note:'field name' },
            { name:'dispatcher', type:'object', required:false, defaultValue:null, note:'flux dispatcher' },
            { name:'boxClass', type:'string', required:false, defaultValue:'', note:'container CSS class' },
            { name:'inputClass', type:'string', required:false, defaultValue:'form-control', note:'input element CSS class' },
            { name:'type', type:'string', required:false, defaultValue:'json', note:'input type' },
            { name:'label', type:'string', required:false, defaultValue:'', note:'input label' },
            { name:'value', type:'string', required:false, defaultValue:'', note:'input type' }
        ];
        return attributes;
    },
    
    getValue: function() {
        return this.state.value;
    },
    
    onChange: function(event) {
        var newValue = event.target.value;
        this.setState({ value:newValue });
        this.fire('change', [newValue]);
    },
    
    render: function() {
        var displayValue = this.state.value;
        // set content display
        var content = 
            React.createElement("div", {className:  "form-group html-text-edit-content-container" }, 
                React.createElement("label", null,  this.state.label), 
                React.createElement("textarea", {
                    name:  this.state.name, 
                    className:  this.state.inputClass, 
                    value:  displayValue, 
                    onChange:  this.onChange}
                )
            );
        return (
            React.createElement("div", {className:  this.state.containerClassNames.join(' ') }, 
                 content 
            )
        );
    }
});

module.exports = HtmlTextEdit;
