// reactlet-html-button/src/html_button.js
/** @jsx React.DOM */

var React = React || require('react/addons');

module.exports = React.createClass({
    name: 'html-button',
    mixins: [getCommonMixin],
    
    // attribute definitions
    getAttributes: function() {
        var attributes = [
            { name:'name', type:'string', required:true, defaultValue:'', note:'field name' },
            { name:'dispatcher', type:'object', required:false, defaultValue:null, note:'flux dispatcher' },
            { name:'boxClass', type:'string', required:false, defaultValue:'', note:'container CSS class' },
            { name:'inputClass', type:'string', required:false, defaultValue:'btn btn-info', note:'CSS class' },
            { name:'type', type:'string', required:false, defaultValue:'submit', note:'button input type' },
            { name:'value', type:'string', required:false, defaultValue:'Submit', note:'input value' }
        ];
        return attributes;
    },
    
    getValue: function() {
        return this.state.value;
    },
    
    /*
    onChange: function(event) {
        var newValue = event.target.value;
        this.setState({ value:newValue });
        this.fire('change', [newValue]);
    },
    */
    
    render: function() {
        // set content display
        var content = 
            <input
                type={ this.state.type }
                name={ this.state.name }
                className={ this.state.inputClass }
                value={ this.state.value }
            />;
        return (
            <div className={ this.state.containerClassNames.join(' ') }  >
                { content }
            </div>
        );
    }
});
