// reactlet-html-input/src/html_input.js
/** @jsx React.DOM */

var React = React || require('react/addons');

module.exports = React.createClass({
    name: 'html-input',
    mixins: [getCommonMixin],
    
    // attribute definitions
    getAttributes: function() {
        var attributes = [
            { name:'name', type:'string', required:true, defaultValue:'', note:'field name' },
            { name:'dispatcher', type:'object', required:false, defaultValue:null, note:'flux dispatcher' },
            { name:'boxClass', type:'string', required:false, defaultValue:'', note:'container CSS class' },
            { name:'inputClass', type:'string', required:false, defaultValue:'form-control', note:'input element CSS class' },
            { name:'type', type:'string', required:false, defaultValue:'text', note:'input type' },
            { name:'label', type:'string', required:false, defaultValue:'', note:'input label' },
            { name:'value', type:'string', required:false, defaultValue:'', note:'input type' },
            { name:'placeholder', type:'string', required:false, defaultValue:'', note:'placeholder text' }
        ];
        return attributes;
    },
    
    getValue: function() {
        return this.state.value;
    },
    
    onChange: function(event) {
        var newValue = event.target.value;
        this.setState({ value:newValue });
        this.fire('change', [newValue]);
    },
    
    render: function() {
        console.log('>>> html_input value:', this.state.value);
        // set content display
        var label = '';
        var currentValue = '';
        if (this.state.type !== 'hidden') {
            label = <label>{ this.state.label }</label>;
        }
        if (this.state.type == 'file' && this.state.value) {
            // this.state.value could be an array or object
            var displayValue = '';
            if (this.state.value  instanceof Array) {
                for (var i = 0; i < this.state.value.length; i++) {
                    displayValue += this.state.value[i].originalname + ';';
                }
            } else {
                displayValue = this.state.value.originalname || '';
            }
            currentValue = <div><label>{ displayValue }</label></div>;
        }
        var content = 
            <div className={ "form-group html-input-content-container" } >
                { label }
                { currentValue }
                <input
                    type={ this.state.type }
                    name={ this.state.name }
                    className={ this.state.inputClass }
                    placeholder={ this.state.placeholder }
                    value={ this.state.value }
                    onChange={this.onChange }
                />
            </div>;
        return (
            <div className={ this.state.containerClassNames.join(' ') }  >
                { content }
            </div>
        );
    }
});
