// reactlet-html-web-edit/src/html_web_edit.js
var React = React || require('react/addons');

module.exports = React.createClass({
    name: 'html-webedit',
    mixins: [getCommonMixin],
    
    // attribute definitions
    getAttributes: function() {
        var attributes = [
            { name:'name', type:'string', required:true, defaultValue:'', note:'field name' },
            { name:'dispatcher', type:'object', required:false, defaultValue:null, note:'flux dispatcher' },
            { name:'boxClass', type:'string', required:false, defaultValue:'', note:'container CSS class' },
            { name:'inputClass', type:'string', required:false, defaultValue:'form-control summernote', note:'input element CSS class' },
            { name:'type', type:'string', required:false, defaultValue:'json', note:'input type' },
            { name:'label', type:'string', required:false, defaultValue:'', note:'input label' },
            { name:'value', type:'string', required:false, defaultValue:'', note:'input type' }
        ];
        return attributes;
    },
    
    getValue: function() {
        return this.state.value;
    },
    
    onChange: function(event) {
        var newValue = event.target.value;
        this.setState({ value:newValue });
        this.fire('change', [newValue]);
    },
    
    componentDidMount: function() {
        var noteElement = this.refs.summernote;
        $(noteElement.getDOMNode()).summernote({
            height: 270
        });
    },
    
    render: function() {
        var displayValue = this.state.value;
        // set content display
        var content = 
            <div className={ "form-group html-web-edit-content-container" } >
                <label>{ this.state.label } (html)</label>
                <textarea
                    ref="summernote"
                    name={ this.state.name }
                    className={ this.state.inputClass }
                    value={ displayValue }
                    onChange={ this.onChange }
                />
            </div>;
        return (
            <div className={ this.state.containerClassNames.join(' ') }  >
                { content }
            </div>
        );
    }
});
